src/archivematicaCommon/lib/archivistsToolkit/client.py

```diff
@@ -235,12 +235,24 @@ def find_collection_ids(self, search_pattern='', page=None, page_size=30):
         """
         cursor = self.db.cursor()
 
-        if search_pattern == '':
+        if search_pattern == '' and identifier == '':
             sql = "SELECT resourceId FROM Resources ORDER BY title"
             params = ()
         else:
-            sql = "SELECT resourceId FROM Resources WHERE (title LIKE %s OR resourceid LIKE %s) AND resourceLevel in ('recordgrp', 'collection') ORDER BY title"
-            params = ('%' + search_pattern + '%', '%' + search_pattern + '%')
+            clause = 'resourceid LIKE %s'
+            params = ['%' + search_pattern + '%']
+
+            if search_pattern != '':
+                clause = 'title LIKE %s OR' + clause
+                params.insert(0, '%' + search_pattern + '%')
+
+            if identifier != '':
+                clause = 'resourceIdentifier1 LIKE %s OR ' + clause
+                params.insert(0, '%' + identifier + '%')
+
+            params = tuple(params)
+
+            sql = "SELECT resourceId FROM Resources WHERE ({}) AND resourceLevel in ('recordgrp', 'collection') ORDER BY title".format(clause)
```

Comment:
The params are SQL placeholders; they're not string interpolation, but rather passing parameters to stored procedures. Since this isn't a single value it can't be passed in the params that way.